diff --git a/samples/bpf/Makefile b/samples/bpf/Makefile
index 1303af10..dca596cd 100644
--- a/samples/bpf/Makefile
+++ b/samples/bpf/Makefile
@@ -204,7 +204,7 @@ all:
 
 clean:
 	$(MAKE) -C ../../ M=$(CURDIR) clean
-	@rm -f *~
+	@rm -f *~ *.bpf
 
 $(LIBBPF): FORCE
 # Fix up variables inherited from Kbuild that tools/ build system won't like
@@ -257,3 +257,4 @@ $(obj)/%.o: $(src)/%.c
 		-Wno-address-of-packed-member -Wno-tautological-compare \
 		-Wno-unknown-warning-option $(CLANG_ARCH_ARGS) \
 		-O2 -emit-llvm -c $< -o -| $(LLC) -march=bpf -filetype=obj -o $@
+	llvm-objdump -S -no-show-raw-insn $@ > ${@:.o=.bpf}
